openxr/examples/opengl_xlib.rs

```diff
@@ -0,0 +1,264 @@
+#[macro_use]
+extern crate glium;
+
+extern crate x11;
+extern crate openxr;
+
+use std::ptr;
+use openxr as xr;
+use x11::{xlib, glx};
+use std::ffi::{c_void, CString};
+use glium::texture::{DepthTexture2d, DepthFormat, UncompressedFloatFormat, MipmapsOption};
+
+const GL_RGBA8: u32 = 0x8058;
+
+pub struct Backend {
+    context: glx::GLXContext,
+    display: *mut xlib::Display,
+    visual: *mut xlib::XVisualInfo,
+    fb_config: glx::GLXFBConfig,
+    drawable: x11::xlib::Drawable
+}
+
+impl Backend {
+    pub fn new() -> Self{
+        let mut fbcount = 0;
+        let visual_attribs = [0];
+
+        let context_attribs = [
+            glx::GLX_RGBA, glx::GLX_DEPTH_SIZE, 24, glx::GLX_DOUBLEBUFFER, 0
+        ];
+        unsafe{
+            let display = xlib::XOpenDisplay(ptr::null());
+            let root = xlib::XDefaultRootWindow(display);
+            let visual = glx::glXChooseVisual(display, 0, context_attribs.as_ptr() as *mut _);
+            let context = glx::glXCreateContext(display, visual, 0 as glx::GLXContext, true as xlib::Bool);
+            let fb_config = glx::glXChooseFBConfig(display, xlib::XDefaultScreen(display), visual_attribs.as_ptr(), &mut fbcount);
+
+            glx::glXMakeCurrent(display, root, context);
+
+            Self{
+                context,
+                display,
+                visual,
+                fb_config: *fb_config,
+                drawable: root
+            }
+        }
+    }
+    pub fn xr_session_create_info(&self) -> xr::opengl::SessionCreateInfo{
```

Comment:
We could provide a cleaner interface by calling `openxr::Instance::create_session` directly here, since this function can (more or less) guarantee the validity of its inputs, thereby resolving most of the safety requirements of `create_session`, though the resulting function should still be marked `unsafe` due to the unchecked lifetimes.